scripts/builtin/msvm.dml

```diff
@@ -42,34 +43,45 @@
 # ---------------------------------------------------------------------------------------------
 # model           Double   ---        model matrix
 
-m_msvm = function(Matrix[Double] X, Matrix[Double] Y, Boolean intercept = FALSE, Integer num_classes =10,
+m_msvm = function(Matrix[Double] X, Matrix[Double] Y, Boolean intercept = FALSE,
                   Double epsilon = 0.001, Double lambda = 1.0, Integer max_iterations = 100, Boolean verbose = FALSE)
   return(Matrix[Double] model)
 {
   if(verbose)
     print("Built-in Multiclass-SVM started")
 
-  num_samples = nrow(X)
-  num_features = ncol(X)
-
-  num_rows_in_w = num_features
+  num_rows_in_w = ncol(X)
   if(intercept) {
     num_rows_in_w = num_rows_in_w + 1
   }
-  
-  w = matrix(0, rows=num_rows_in_w, cols=num_classes)
 
-  parfor(iter_class in 1:num_classes) {
-    Y_local = 2 * (Y == iter_class) - 1
-    if(verbose) {
-      print("iter class: " + iter_class)
-      print("y local: " + toString(Y_local))
+  if(ncol(Y) > 1){
+    w = matrix(0, rows=num_rows_in_w, cols=ncol(Y))
+    # If multiple columns, assume the input to be OneHotEncoded already.
+    if((min(Y) != 0 | min(Y) != -1) & max(Y) != 1)
+      stop("MSVM: Invalid OneHot input matrix Y")
+    
+    parfor(class in 1:ncol(Y)){
+      w[,class] = l2svm(X=X, Y=Y[,class], intercept=intercept,
+          epsilon=epsilon, lambda=lambda, max_iterations=max_iterations, 
+          verbose= verbose, column_id=class)
     }
```

Comment:
Instead of replicating this code of running the core of msvm, please simply modify Y as follows:
```
if( ncol(Y) > 1 )
  Y = rowMax(Y * t(seq(1,ncol(Y))))
```